prereise/gather/demanddata/transportation_electrification/immediate.py

```diff
@@ -208,33 +208,107 @@ def immediate_charging(
     flag_translation = {1: "weekend", 2: "weekday"}
     for i, weekday_flag in enumerate(input_day):
         daily_profile = daily_resampled_profiles[flag_translation[weekday_flag]]
-        if i == len(input_day) - 1:
-            # for the last day of the year, we need to wrap our 48 hours back around
-            # first half: last day of the year
-            model_year_profile[-24:] += daily_profile[:24]
-            # last half: first day of the year
-            model_year_profile[:24] += daily_profile[24:]
-        else:
-            model_year_profile[i * 24 : i * 24 + 48] += daily_profile
+
+        # create wrap-around indexing function
+        trip_window_indices = np.arange(i * 24, i * 24 + 72) % len(model_year_profile)
+
+        # MW
+        model_year_profile[trip_window_indices] += daily_profile
+
     # Normalize the output so that it sums to 1
     summed_profile = model_year_profile / model_year_profile.sum()
 
     return summed_profile
 
 
-def adjust_bev(hourly_profile, adjustment_values):  # noqa: N802
+def adjust_bev(
+    hourly_profile,
+    adjustment_values,
+    model_year,
+    veh_type,
+    veh_range,
+    bev_vmt,
+    charging_efficiency,
+):  # noqa: N802
     """Adjusts the charging profiles by applying weighting factors based on
     seasonal/monthly values
 
     :param numpy.ndarray hourly_profile: normalized charging profiles
     :param pandas.DataFrame adjustment_values: weighting factors for each
         day of the year loaded from month_info_nhts.mat.
-    :return: (*numpy.ndarray*) -- the final adjusted charging profiles.
+    :param int model_year: year that is being modelled/projected to, 2017, 2030, 2040, 2050.
+    :param str veh_type: determine which category (MDV or HDV) to produce charging profiles for
+    :param int veh_range: 100, 200, or 300, represents how far vehicle can travel on single charge.
+    :param int/float bev_vmt: BEV VMT value / scaling factor loaded from Regional_scaling_factors.csv
+    :param float charging_efficiency: from grid to battery efficiency.
+    :return: (*numpy.ndarray*) -- final adjusted charging profiles.
+    """
+    kwhmi = data_helper.get_kwhmi(model_year, veh_type, veh_range)
+
+    # weekday/weekend, monthly urban and rural moves scaling
+    adjusted_load = apply_daily_adjustments(
+        hourly_profile,
+        adjustment_values,
+    )
+
+    # simulation year urban and rural scaling specific to region
+    simulation_hourly_profile = apply_annual_scaling(
+        adjusted_load,
+        bev_vmt,
+        charging_efficiency,
+        kwhmi,
+    )
+
+    return simulation_hourly_profile
+
+
+def apply_daily_adjustments(
+    hourly_profile,
+    adjustment_values,
+    num_days_per_year=365,
+    num_segments_per_day=24,
+):  # noqa: N802
+    """Adjusts the charging profiles by applying weighting factors based on
+    seasonal/monthly values
```

Comment:
Should the function description be more specific so that it differentiates what the three functions do here?